smart-contract-verifier/src/solidity/client.rs

```diff
@@ -3,12 +3,24 @@ use crate::{compiler::Compilers, middleware::Middleware, verifier::Success};
 use std::sync::Arc;
 
 pub struct ClientBuilder {
-    compilers: Compilers<SolidityCompiler>,
+    compilers: Arc<Compilers<SolidityCompiler>>,
     middleware_stack: Vec<Arc<dyn Middleware<Success>>>,
 }
 
 impl ClientBuilder {
+    /// Convenience method to initialize new client builder.
+    ///
+    /// If you need to keep a reference to the compilers after initialization, use [`new_arc`].
+    ///
+    /// [`new_arc`]: Self::new_arc
     pub fn new(compilers: Compilers<SolidityCompiler>) -> Self {
+        Self::new_arc(Arc::new(compilers))
+    }
+
+    /// Initialize new client builder. [`new`] is more ergonomic if you don't need the `Arc`.
+    ///
+    /// [`new`]: Self::new
+    pub fn new_arc(compilers: Arc<Compilers<SolidityCompiler>>) -> Self {
```

Comment:
but... why... is it really to spare `Arc::new(compilers)` when calling `new`?

Reply:
why do we need this function? it's just the same as `new(Arc::new(compilers))`

Reply:
`new(Compilers<T>)` is written for convenience purpose, as for the most initializations users do not care about keeping a reference to compilers. `new_arc(Arc<Compilers<T>>)` is used for those cases when that reference is important (currently it is used only in tests and probably we may add a `#[cfg(test)]` directive, but I personally don't see why can't be a part of the builder for usual cases as well)